Magento_Swatches/web/js/swatch-renderer.js

```diff
@@ -708,14 +702,14 @@ define([
               $parent.removeAttr('option-selected').find('.selected').removeClass('selected');
               $input.val('');
               $label.text('');
-              $option.attr('aria-checked', false);
+              $this.attr('aria-checked', false);
           } else {
-              $parent.attr('option-selected', $option.attr('option-id')).find('.selected').removeClass('selected');
-              $label.text($option.attr('option-label'));
-              $input.val($option.attr('option-id'));
+              $parent.attr('option-selected', $this.attr('option-id')).find('.selected').removeClass('selected');
+              $label.text($this.attr('option-label'));
+              $input.val($this.attr('option-id'));
               $input.attr('data-attr-name', this._getAttributeCodeById(attributeId));
               $this.addClass('selected');
-              $widget._toggleCheckedAttributes($option, $wrapper);
+              $widget._toggleCheckedAttributes($this, $wrapper);
           }
```

Comment:
Generally it works quite well and aria attributes are applicable where it should, I found 2 issues:
1. when option is selected the value of `aria-checked` on `.swatch__option` element change but on `.swatch__container selected` `aria-checked="false"`. I think `aria-checked` from the `.swatch__container selected` should be removed completely
2. It'll be better to use `aria-selected` instead of `aria-checked` it is possible in Magento with no rewriting whole core things?
3. Naming is confusing a little: swatch__container is not a swatch container but swatch option container... maybe we can update this?

Reply:
Ok, I replaced `aria-checked` with `aria-selected` and changed swatch container class to `swatch__option-container`